<a href="https://colab.research.google.com/github/javierexe/estimacion-de-cosecha/blob/main/Proyecto_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Proyecto 2. Predicción de rendimientos

#Parte 1

##Primera opción: Datos Llayquén

> Diccionario de datos

| **Nombre**                | **Descripción**                                                         | **Tipo** |
|----------------------------|-------------------------------------------------------------------------|----------|
|          Estanque          | Caseta de riego                                                         | Object   |
|           Unidad           | Cabezal de riego                                                        | Object      |
|           Sector           | Sector de riego o cuartel                                               | Object   |
|       Superficie (ha)      | Tamaño del sector de riego (ha)                                         | Float    |
| Superficie geometrica (ha) | Tamaño del polígono del sector de riego (ha). Dibujado en Google Earth  | Float    |
|      Marco plantación      | Distancia de plantación (m)                                             | Object   |
|             Año            | Año de plantación del sector                                            | Date     |
|         Temporada          | Temporada de producción desde agosto a julio                            | Object   |
|          Nitrógeno         | Cantidad de Nitrógeno aportado con la fertilización (kg)                | Int      |
|           Potasio          | Cantidad de Potasio aportado con la fertilización (kg)                  | Int      |
|             Eto            | Evapotrasnpiración potencial en la temporadav(mm)                       | Int      |
|             m3             | Volumen devriego aplicado a cada sector en la temporada (m3)            | Int      |
|         días grado         | Días grado base 12                                                      | Int      |
|         rendimiento        | Rendimiento de cada cuartel por temporada (ton/ha)                      | Float    |

---








1) Fuente de datos
  
  > ***R.: Los datos provienen del área técnica de la empresa agrícola Jorge Schmidt y Cía. Ltda.***


2) Breve descripción de datos

  > ***R.: Los datos corresponden a las estadíticas productivas de un campo de  paltos, las que están acompañadas de algunas variables que influyen en el resultado productivo de cada temporada, tales como riego, fertilización, variables climáticas, año de plantación, entre otras.***

3) ¿Cuál es el objetivo?

>  ***R.: El objetivo es construir un modelo que ayude a predecir el rendimiento  que alcanzará cada cuartel la próxima temporada.***

4) ¿Este es un problema de clasificación o regresión?

>  ***R.: Es un problema de regresión***

5) ¿Cuántas características hay?

>  ***R.: Hay 13 características***

6) ¿Cuántas filas de datos hay?

>  ***R.: Hay 550 filas***

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

>  ***R.: Por la naturaleza de la fuente de los datos es probable que hayan datos mal digitados o erróneos, datos faltantes.***

> ***Hay que generar algunas columnas calculadas en base a otras.***

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
import pandas as pd
df = pd.read_excel('/content/drive/MyDrive/Coding DOJO/files/datos Llayquen.xlsx')
df.head(15)

Estanque  Unidad Sector  Superficie (ha)  Superficie geometrica (ha)  \
0         1       1      1             4.85                         6.5   
1         1       1      2             3.20                         3.6   
2         1       1      3             6.26                         6.5   
3         1       1     4a             2.80                         3.1   
4         1       1     4b             1.80                         2.1   
5         1       1    4.1             1.70                         2.1   
6         1       1    5Fr             3.03                         4.8   
7         1       2      1             5.86                         6.0   
8         1       2      2             9.60                         9.6   
9         1       2      3             5.40                         6.1   
10        1       2     4a             3.70                         4.4   
11        1       2     4b             1.70                         2.0   
12        1       2    4.1             2.60                         4.5   
13        1       2    5Fr             3.00                         3.2   
14        2       3      1             0.42                         0.6   

   Marco plantación  Año plantación Temporada   Nitrógeno     Potasio  Eto  \
0             8 x 3            2003   2017/18  265.371429  173.061224  NaN   
1             8 x 3            2003   2017/18  275.450000  208.543750  NaN   
2             8 x 3            2003   2017/18  292.185714  200.546032  NaN   
3             6 x 2            2003   2017/18  188.800000  196.282353  NaN   
4             6 x 2            2003   2017/18  271.100000   44.357143  NaN   
5             6 x 2            2003   2017/18  251.000000  210.763636  NaN   
6             6 x 2            2003   2017/18  332.000000  203.000000  NaN   
7             8 x 3            2003   2017/18  274.835593  201.000000  NaN   
8             8 x 3            2003   2017/18  229.143750    0.000000  NaN   
9             8 x 3            2003   2017/18  250.025926    0.000000  NaN   
10            6 x 2            2003   2017/18  283.800000  205.000000  NaN   
11            6 x 2            2003   2017/18  229.000000  196.000000  NaN   
12            6 x 2            2003   2017/18  242.000000    0.000000  NaN   
13            6 x 2            2003   2017/18  333.400000  239.000000  NaN   
14            7 x 2            2003   2017/18   53.900000    0.000000  NaN   

    m3  días grado  rendimiento  
0  NaN         NaN         19.2  
1  NaN         NaN         22.3  
2  NaN         NaN         22.1  
3  NaN         NaN          6.8  
4  NaN         NaN         26.6  
5  NaN         NaN         14.8  
6  NaN         NaN         32.1  
7  NaN         NaN         13.7  
8  NaN         NaN         19.4  
9  NaN         NaN         24.3  
10 NaN         NaN         18.8  
11 NaN         NaN         23.5  
12 NaN         NaN         17.0  
13 NaN         NaN         43.0  
14 NaN         NaN         38.3

In [3]:
df.shape

(550, 14)

In [4]:
df.describe()

Unidad  Superficie (ha)  Superficie geometrica (ha)  \
count  550.000000       550.000000                  550.000000   
mean     6.854545         1.658063                    1.860909   
std      3.529311         1.528198                    1.677511   
min      1.000000         0.300000                    0.300000   
25%      4.000000         0.726250                    0.800000   
50%      7.000000         1.000000                    1.200000   
75%     10.000000         2.000000                    2.200000   
max     12.000000         9.600000                    9.600000   

       Año plantación   Nitrógeno     Potasio          Eto             m3  \
count      550.000000  550.000000  533.000000   440.000000     440.000000   
mean      2004.781818  277.982019  224.738321  1318.043375   23293.069886   
std          1.915243   94.482907   72.579755    34.822912   18641.264297   
min       2003.000000   27.789474    0.000000  1274.775000    3767.000000   
25%       2003.000000  239.025000  190.652174  1299.888750   11888.500000   
50%       2005.000000  290.244444  219.377778  1312.885000   18143.000000   
75%       2005.000000  330.532955  255.550000  1331.039625   24868.750000   
max       2008.000000  556.008333  564.133333  1371.628500  114392.000000   

        días grado  rendimiento  
count   440.000000   550.000000  
mean   1511.000000    20.683513  
std     251.505988    10.246509  
min    1116.000000     0.000000  
25%    1394.250000    13.219366  
50%    1572.000000    21.100000  
75%    1688.750000    27.950000  
max    1784.000000    46.600000

##Segunda opción: Spotify Tracks Dataset

| **Column** | **Description** |  |
|---|---|---|
| track_id | The Spotify ID for the track |  |
| artists | The artists' names who performed the track. If there is more than one artist, they are separated by a ; |  |
| album_name  | The album name in which the track appears |  |
| track_name | Name of the track |  |
| popularity | The popularity of a track is a value between 0 and 100, with 100 being the most popular.  |  |
|  | The popularity is calculated by algorithm and is based, in the most part, on the total number  |  |
|  | of plays the track has had and how recent those plays are. Generally speaking, songs that are  |  |
|  | being played a lot now will have a higher popularity than songs that were played a lot in the past.  |  |
|  | Duplicate tracks (e.g. the same track from a single and an album) are rated independently.  |  |
|  | Artist and album popularity is derived mathematically from track popularity. |  |
| duration_ms | The track length in milliseconds |  |
| explicit | Whether or not the track has explicit lyrics (true = yes it does; false = no it does not OR unknown) |  |
| danceability | Danceability describes how suitable a track is for dancing based on a combination of musical elements including tempo,  |  |
|  | rhythm stability, beat strength, and overall regularity. A value of 0.0 is least danceable and 1.0 is most danceable |  |
| energy | Energy is a measure from 0.0 to 1.0 and represents a perceptual measure of intensity and activity. Typically, energetic  |  |
|  | tracks feel fast, loud, and noisy. For example, death metal has high energy, while a Bach prelude scores low on the scale |  |
| key | The key the track is in. Integers map to pitches using standard Pitch Class notation. E.g. 0 = C, 1 = C♯/D♭, 2 = D, and so on. If no key was detected, the value is -1 |  |
| loudness  | The overall loudness of a track in decibels (dB) |  |
| mode | Mode indicates the modality (major or minor) of a track, the type of scale from which its melodic content is derived. Major is represented by 1 and minor is 0 |  |
| speechiness | Speechiness detects the presence of spoken words in a track. The more exclusively speech-like the recording (e.g. talk show, audio book, poetry), the closer to 1.0  |  |
|  | the attribute value. Values above 0.66 describe tracks that are probably made entirely of spoken words. Values between 0.33 and 0.66 describe tracks that may contain  |  |
| acousticness | A confidence measure from 0.0 to 1.0 of whether the track is acoustic. 1.0 represents high confidence the track is acoustic |  |
| instrumentalness | Predicts whether a track contains no vocals. "Ooh" and "aah" sounds are treated as instrumental in this context. Rap or spoken word tracks are clearly "vocal".  |  |
|  | The closer the instrumentalness value is to 1.0, the greater likelihood the track contains no vocal content |  |
| liveness | Detects the presence of an audience in the recording. Higher liveness values represent an increased probability that the track was performed live.  |  |
|  | A value above 0.8 provides strong likelihood that the track is live |  |
| valence | A measure from 0.0 to 1.0 describing the musical positiveness conveyed by a track. Tracks with high valence sound more positive (e.g. happy, cheerful, euphoric),  |  |
|  | while tracks with low valence sound more negative (e.g. sad, depressed, angry) |  |
| tempo | The overall estimated tempo of a track in beats per minute (BPM). In musical terminology, tempo is the speed or pace of a given piece and derives directly from the average beat duration |  |
| time_signature | An estimated time signature. The time signature (meter) is a notational convention to specify how many beats are in each bar (or measure).  |  |
|  | The time signature ranges from 3 to 7 indicating time signatures of 3/4, to 7/4. |  |
| track_genre | The genre in which the track belongs |  |

1) Fuente de datos
  
  > ***R.: Los datos corresponden a un dataset de Kaggle y pueden obtenerse [aquí](https://www.kaggle.com/datasets/maharshipandya/-spotify-tracks-dataset).
  Open Database License (ODbL)

2) Breve descripción de datos

  > ***R.: Este es un conjunto de datos de pistas de Spotify en una gama de 125 géneros diferentes. Cada pista tiene algunas características de audio asociadas.***

3) ¿Cuál es el objetivo?

>  ***R.: Construir un sistema de recomendación basado en alguna entrada o preferencia del usuario***

4) ¿Este es un problema de clasificación o regresión?

>  ***R.: Es un problema de clasificación***

5) ¿Cuántas características hay?

>  ***R.: Hay 21 características***

6) ¿Cuántas filas de datos hay?

>  ***R.: Hay 114.000 filas***

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

> ***R.: Parece ser un data set limpio, sin valores faltantes.***

> ***Hay que trabajar con modelos que áun no vemos***

In [5]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [6]:
df2 = pd.read_csv('/content/drive/MyDrive/Coding DOJO/files/dataset.csv')
df2.head()

Unnamed: 0                track_id                 artists  \
0           0  5SuOikwiRyPMVoIQDJUgSV             Gen Hoshino   
1           1  4qPNDBW1i3p13qLCt0Ki3A            Ben Woodward   
2           2  1iJBSr7s7jYXzM8EGcbK5b  Ingrid Michaelson;ZAYN   
3           3  6lfxq3CG4xtTiEg7opyCyx            Kina Grannis   
4           4  5vjLSffimiIP26QG5WcN2K        Chord Overstreet   

                                          album_name  \
0                                             Comedy   
1                                   Ghost (Acoustic)   
2                                     To Begin Again   
3  Crazy Rich Asians (Original Motion Picture Sou...   
4                                            Hold On   

                   track_name  popularity  duration_ms  explicit  \
0                      Comedy          73       230666     False   
1            Ghost - Acoustic          55       149610     False   
2              To Begin Again          57       210826     False   
3  Can't Help Falling In Love          71       201933     False   
4                     Hold On          82       198853     False   

   danceability  energy  ...  loudness  mode  speechiness  acousticness  \
0         0.676  0.4610  ...    -6.746     0       0.1430        0.0322   
1         0.420  0.1660  ...   -17.235     1       0.0763        0.9240   
2         0.438  0.3590  ...    -9.734     1       0.0557        0.2100   
3         0.266  0.0596  ...   -18.515     1       0.0363        0.9050   
4         0.618  0.4430  ...    -9.681     1       0.0526        0.4690   

   instrumentalness  liveness  valence    tempo  time_signature  track_genre  
0          0.000001    0.3580    0.715   87.917               4     acoustic  
1          0.000006    0.1010    0.267   77.489               4     acoustic  
2          0.000000    0.1170    0.120   76.332               4     acoustic  
3          0.000071    0.1320    0.143  181.740               3     acoustic  
4          0.000000    0.0829    0.167  119.949               4     acoustic  

[5 rows x 21 columns]

In [7]:
df2.shape

(114000, 21)

# Parte 2

Al usar el conjunto de datos aprobado que propusieron para el proyecto 2, parte 1 la semana pasada, limpiarán y explorarán el conjunto de datos.  

Para cumplir con los requisitos para esta semana, deben haber:

- eliminado las columnas innecesarias

- eliminado las filas duplicadas

- identificado y abordado los valores faltantes

- identificado y corregido las inconsistencias en los datos para los valores categóricos (es decir, Cat, cat, cats)

- producido visuales univariantes para el objetivo y todas las características

- identificado los valores atípicos

- producido un mapa de calor de correlación e identificado cualquier correlaciones moderadas o fuertes entre el objetivo y las características y/o entre una característica y otra

- comentado todos los pasos de limpieza y descrito todas las decisiones que tomaron. (Nota: durante la semana final del programa, harán una presentación para una audiencia técnica que incluirá sus decisiones y procesos en los pasos de limpieza. Entre más notas tomen ahora, más fácil será la presentación. Para clarificar, esa presentación NO se entrega ahora).

**Paso siguiente:**
>Traemos el diccionario de datos

>Cargaremos los datos

> Diccionario de datos

| **Nombre**                | **Descripción**                                                         | **Tipo** |
|----------------------------|-------------------------------------------------------------------------|----------|
|          Estanque          | Caseta de riego                                                         | Object   |
|           Unidad           | Cabezal de riego                                                        | Object      |
|           Sector           | Sector de riego o cuartel                                               | Object   |
|       Superficie (ha)      | Tamaño del sector de riego (ha)                                         | Float    |
| Superficie geometrica (ha) | Tamaño del polígono del sector de riego (ha). Dibujado en Google Earth  | Float    |
|      Marco plantación      | Distancia de plantación (m)                                             | Object   |
|             Año            | Año de plantación del sector                                            | Date     |
|         Temporada          | Temporada de producción desde agosto a julio                            | Object   |
|          Nitrógeno         | Cantidad de Nitrógeno aportado con la fertilización (kg)                | Float      |
|           Potasio          | Cantidad de Potasio aportado con la fertilización (kg)                  | Float      |
|             Eto            | Evapotrasnpiración potencial en la temporadav(mm)                       | Float      |
|             m3             | Volumen devriego aplicado a cada sector en la temporada (m3)            | Float      |
|         días grado         | Días grado base 12                                                      | Float      |
|         rendimiento        | Rendimiento de cada cuartel por temporada (ton/ha)                      | Float    |

---

In [8]:
df = pd.read_excel('/content/drive/MyDrive/Coding DOJO/files/datos Llayquen.xlsx')
df.head()

Estanque  Unidad Sector  Superficie (ha)  Superficie geometrica (ha)  \
0        1       1      1             4.85                         6.5   
1        1       1      2             3.20                         3.6   
2        1       1      3             6.26                         6.5   
3        1       1     4a             2.80                         3.1   
4        1       1     4b             1.80                         2.1   

  Marco plantación  Año plantación Temporada   Nitrógeno     Potasio  Eto  m3  \
0            8 x 3            2003   2017/18  265.371429  173.061224  NaN NaN   
1            8 x 3            2003   2017/18  275.450000  208.543750  NaN NaN   
2            8 x 3            2003   2017/18  292.185714  200.546032  NaN NaN   
3            6 x 2            2003   2017/18  188.800000  196.282353  NaN NaN   
4            6 x 2            2003   2017/18  271.100000   44.357143  NaN NaN   

   días grado  rendimiento  
0         NaN         19.2  
1         NaN         22.3  
2         NaN         22.1  
3         NaN          6.8  
4         NaN         26.6

---
**Paso sig.:**
>Revisamos si hay valores duplicados en los datos

In [19]:
df.duplicated().sum()

0

**Obs.:**
>No hay valores duplicados

---
**Paso sig.:**
>Revisamos si hay valores nulos en las columnas

In [43]:
print(f'Valores nulos por columna\n'
,'--'*12)
print(df.isnull().sum())
print(f'\nTotal nulos:',df.isnull().sum().sum())

Valores nulos por columna
 ------------------------
Id           0
sup          0
sup_geo      0
dist         0
anio         0
temp         0
N            0
K           17
Eto        110
m3         110
dg         110
rend         0
dtype: int64

Total nulos: 347


**Obs.:**
>Tenemos un total de 347 valores nulos. 17 en la columna *K* y 110 en cada columna de variables climáticas (*Eto, m3 y dg*).

>Más adelante veremos cómo abordar estos valores nulos

---
**Paso sig.:**
>Vamos a ajustar el nombre de las columnas a un formato más adecuado para continuar trabajando con ellos

In [9]:
df.rename(columns={'Superficie (ha)':'sup',
                   'Superficie geometrica (ha)':'sup_geo',
                   'Marco plantación':'dist',
                   'Año plantación':'anio',
                   'Temporada':'temp',
                   'Nitrógeno':'N',
                   'Potasio':'K',
                   'días grado':'dg',
                   'rendimiento':'rend'},
               inplace=True)

**Ref.:** [Analytics Lane](https://www.analyticslane.com/2019/05/06/como-cambiar-el-nombre-de-las-columnas-en-pandas/) ¿Cómo cambiar el nombre de las columnas en Pandas?

In [10]:
df.columns

Index(['Estanque', 'Unidad', 'Sector', 'sup', 'sup_geo', 'dist', 'anio',
       'temp', 'N', 'K', 'Eto', 'm3', 'dg', 'rend'],
      dtype='object')

---
**Paso sig.:**
>Vamos a crear una columna código del sector  de riego y/o cuartel, que corresponde a la unidad productiva a la que hacen referencia estos datos. La denominaremos 'Id'

In [11]:
df['Id'] = df['Estanque'].astype(str) + '_' + df['Unidad'].astype(str) + '_' + df['Sector'].astype(str)
df

Estanque  Unidad Sector       sup  sup_geo       dist  anio     temp  \
0          1       1      1  4.850000      6.5      8 x 3  2003  2017/18   
1          1       1      2  3.200000      3.6      8 x 3  2003  2017/18   
2          1       1      3  6.260000      6.5      8 x 3  2003  2017/18   
3          1       1     4a  2.800000      3.1      6 x 2  2003  2017/18   
4          1       1     4b  1.800000      2.1      6 x 2  2003  2017/18   
..       ...     ...    ...       ...      ...        ...   ...      ...   
545        6      12      8  0.379375      0.5  2,5 x 2,5  2008  2021/22   
546        6      12      9  0.600000      1.1  2,5 x 2,5  2008  2021/22   
547        6      12     10  0.803125      0.8  2,5 x 2,5  2008  2021/22   
548        6      12     11  0.878125      1.2  2,5 x 2,5  2008  2021/22   
549        6      12     12  1.200000      1.2  2,5 x 2,5  2008  2021/22   

              N           K        Eto       m3      dg  rend       Id  
0    265.371429  173.061224        NaN      NaN     NaN  19.2    1_1_1  
1    275.450000  208.543750        NaN      NaN     NaN  22.3    1_1_2  
2    292.185714  200.546032        NaN      NaN     NaN  22.1    1_1_3  
3    188.800000  196.282353        NaN      NaN     NaN   6.8   1_1_4a  
4    271.100000   44.357143        NaN      NaN     NaN  26.6   1_1_4b  
..          ...         ...        ...      ...     ...   ...      ...  
545  444.750000  202.950000  1371.6285   6954.0  1116.0  35.9   6_12_8  
546  304.565476  166.885714  1371.6285  11089.0  1116.0  35.1   6_12_9  
547  298.100000  264.250000  1371.6285  15078.0  1116.0  35.1  6_12_10  
548  289.177778  219.377778  1371.6285  18134.0  1116.0  28.2  6_12_11  
549  219.633333  204.233333  1371.6285  16695.0  1116.0  28.2  6_12_12  

[550 rows x 15 columns]

**Ref.:** [OrigenSendero](https://www.sourcetrail.com/es/python/combine-int-and-object-columns-into-one/) Combine las columnas int y object en una

---
**Paso sig.:**
>Reordenamos de dataframe trayendo la columna 'Id' al principio

>Eliminamos las columnas 'Estanque', 'Unidad' y 'Sector'

In [12]:
col = df.pop('Id')
df.insert(loc= 0 , column= 'Id', value= col)
df.drop(columns=['Estanque', 'Unidad', 'Sector'], inplace=True)

**Ref.:** [DelfStack](https://www.delftstack.com/es/howto/python-pandas/pandas-move-column-to-front/#google_vignette) Mover columna al frente en Pandas DataFrame

In [13]:
df.head()

Id   sup  sup_geo   dist  anio     temp           N           K  Eto  \
0   1_1_1  4.85      6.5  8 x 3  2003  2017/18  265.371429  173.061224  NaN   
1   1_1_2  3.20      3.6  8 x 3  2003  2017/18  275.450000  208.543750  NaN   
2   1_1_3  6.26      6.5  8 x 3  2003  2017/18  292.185714  200.546032  NaN   
3  1_1_4a  2.80      3.1  6 x 2  2003  2017/18  188.800000  196.282353  NaN   
4  1_1_4b  1.80      2.1  6 x 2  2003  2017/18  271.100000   44.357143  NaN   

   m3  dg  rend  
0 NaN NaN  19.2  
1 NaN NaN  22.3  
2 NaN NaN  22.1  
3 NaN NaN   6.8  
4 NaN NaN  26.6

>Redefinimos el diccionario de datos



---



| **Nombre**                | **Descripción**                                                         | **Tipo** |
|----------------------------|-------------------------------------------------------------------------|----------|
|          Id          | Código del sector de riego y/o cuartel                                                         | Object   |
|       Sup       | Tamaño del sector de riego (ha)                                         | Float    |
| Sup_geo | Tamaño del polígono del sector de riego (ha). Dibujado en Google Earth  | Float    |
|      dist      | Distancia de plantación (m)                                             | Object   |
|             Anio            | Año de plantación del sector                                            | Date     |
|         temp          | Temporada de producción desde agosto a julio                            | Object   |
|          N          | Cantidad de Nitrógeno aportado con la fertilización (kg)                | Float      |
|           K          | Cantidad de Potasio aportado con la fertilización (kg)                  | Float      |
|             Eto            | Evapotrasnpiración potencial en la temporadav(mm)                       | Float      |
|             m3             | Volumen devriego aplicado a cada sector en la temporada (m3)            | Float      |
|         dg         | Días grado base 12                                                      | Float      |
|         rend        | Rendimiento de cada cuartel por temporada (ton/ha)                      | Float    |

---

In [14]:
df.dtypes

Id          object
sup        float64
sup_geo    float64
dist        object
anio         int64
temp        object
N          float64
K          float64
Eto        float64
m3         float64
dg         float64
rend       float64
dtype: object

**Obs.:**
>Vemos que el tipo de datos de todas las columnas está deacuerdo al diccionario



---


**Paso sig.:**
>Revisemos si hay valores inconsistentes en las columnas categóricas

In [61]:
df['Id'].nunique()

110

**Obs.**
>Los datos corresponden a los rendimientos productivos de 110 sectores de riego

>No hay valores inconsistentes en la columna **Id**

In [62]:
df['dist'].value_counts()

3 x 3        190
7 x 2        170
2,5 x 2,5    120
6 x 2         40
8 x 3         30
Name: dist, dtype: int64

**Obs.**
>Hay registros de 5 marcos o distancias de plantación disntintas en la columna.

>No hay valores inconsistentes en **dist**.

In [63]:
df['temp'].value_counts()

2017/18    110
2018/19    110
2020/19    110
2020/21    110
2021/22    110
Name: temp, dtype: int64

>Hay registros de 5 temporadas productivas distintas en la columna, desde 2017 a 2022.

>No hay valores inconsistentes en **temp**.